src/v2/computeBidsAsks.ts

```diff
@@ -0,0 +1,78 @@
+import { BigNumber } from '@uniswap/sdk'
+
+function getAmountOut(
+  amountIn: BigNumber,
+  reservesIn: BigNumber,
+  reservesOut: BigNumber
+): { amountOut: BigNumber; reservesInAfter: BigNumber; reservesOutAfter: BigNumber } {
+  const amountOut = amountIn.eq(0)
+    ? new BigNumber(0)
+    : reservesOut.minus(reservesOut.multipliedBy(reservesIn).dividedBy(reservesIn.plus(amountIn.multipliedBy(0.997))))
+  return {
+    amountOut,
+    reservesInAfter: reservesIn.plus(amountIn),
+    reservesOutAfter: reservesOut.minus(amountOut)
+  }
+}
+
+function getAmountIn(
+  amountOut: BigNumber,
+  reservesIn: BigNumber,
+  reservesOut: BigNumber
+): { amountIn: BigNumber; reservesInAfter: BigNumber; reservesOutAfter: BigNumber } {
+  const amountIn = amountOut.eq(0)
+    ? new BigNumber(0)
+    : amountOut.isGreaterThanOrEqualTo(reservesOut)
+    ? new BigNumber(Infinity)
+    : reservesIn
+        .multipliedBy(reservesOut)
+        .dividedBy(reservesOut.minus(amountOut)) // reserves in after
+        .minus(reservesIn) // minus reserves in
+        .dividedBy(0.997) // fee
+
+  return {
+    amountIn,
+    reservesInAfter: reservesIn.plus(amountIn),
+    reservesOutAfter: reservesOut.minus(amountOut)
+  }
+}
+
+export function computeBidsAsks(
+  baseReserves: BigNumber,
+  quoteReserves: BigNumber,
+  numSegments: number = 20
+): { bids: [string, string][]; asks: [string, string][] } {
+  if (baseReserves.eq(0) || quoteReserves.eq(0)) {
+    return {
+      bids: [],
+      asks: []
+    }
+  }
+
+  const increment = baseReserves.dividedBy(numSegments)
+  const baseAmounts = Array.from({ length: numSegments }, (x, i): BigNumber => increment.multipliedBy(i))
```

Comment:
```suggestion
  const baseAmounts = Array.from({ length: numSegments - 1 }, (x, i): BigNumber => increment.multipliedBy(i))
```

this would make it so we're never trying to calculate inputs/outputs that drain the entirety of the pool in a direction, which is probably desirable